SnoopCompileCore/src/snoopi_deep.jl

```diff
@@ -70,17 +74,34 @@ function addchildren!(parent::InferenceTimingNode, t::Core.Compiler.Timings.Timi
     end
 end
 
-function start_deep_timing()
-    Core.Compiler.Timings.reset_timings()
-    Core.Compiler.__set_measure_typeinf(true)
-end
-function stop_deep_timing()
-    Core.Compiler.__set_measure_typeinf(false)
-    Core.Compiler.Timings.close_current_timer()
-end
-
-function finish_snoopi_deep()
-    return InferenceTimingNode(Core.Compiler.Timings._timings[1])
+if have_clear_and_fetch_timings
+    function start_deep_timing()
+        Core.Compiler.__set_measure_typeinf(true)
+    end
+    function stop_deep_timing()
+        Core.Compiler.__set_measure_typeinf(false)
+    end
+    function finish_snoopi_deep()
+        # Construct a dummy Timing for ROOT(), for backwards compatibility with the old API.
+        root = Timing(
+            InferenceFrameInfo(ROOTmi, 0x0, Any[], Any[Core.Const(ROOT)], 1),
+            0x0)
+        root.children = Core.Compiler.Timings.clear_and_fetch_timings()
+
+        return InferenceTimingNode(root)
```

Comment:
@timholy: In https://github.com/JuliaLang/julia/pull/47615, we changed the API to return a _vector of inference trees_, rather than a single inference tree with the dummy "ROOT" node.

We think this is a more sensical API, since there really _wasn't_ a single tree of type inference, there were many, and it also made it easier to get the thread safety correct.

However, of course this breaks the old API. So for now, I'm proposing we just do this, and shove it back into a dummy ROOT node to make a single tree 😅 

Then we can investigate changes to the API in the future if we want to clean things up to all operate on vectors of trees, or just leave it like this forever. :)

Does that sound good to you?